src/kerchunk/gran_to_cat.py

```diff
@@ -0,0 +1,197 @@
+import os
```

Comment:
I prefer sticking with `os` because in this case I am just using `os.path.basename` to extract the filename (with the extension) from the granule URL. But IIRC, doing the same thing is two lines with `pathlib`:

```python
path = pathlib.Path("foo/bar.nc")
basename = path.stem + path.suffix # bar.nc
```

Reply:
@agoodm you can do it with a single line:

```python
name = Path('foo/bar.nc').name
```

Dumping strings to file is also a single line:
```python
out_path = Path.cwd() / name
out_path.with_suffix('.refs.json').write_text(json.dumps(refs, indent=2))
out_path.with_suffix('.stac.json').write_text(json.dumps(stac, indent=2))
```
notably, this strips the current extension and replaces it with the provided one.

And for parsing:
```python
stac = json.loads(Path('bar.stac.json').read_text())
```

Reply:
Fair enough, the 2nd example in particular looks like a compelling enough reason to switch.